### PR TITLE
pyenv 1.0.0

### PR DESCRIPTION
**Pull request by yyuu**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/pyenv.rb [hunk 1]
```diff
@@ -1,9 +1,30 @@
...
+        end
+      elsif DATETIME_VERSIONS =~ other
+        1
+      else
+        super
+      end
+    end
+  end
+
+  pyenv_version = "1.0.0"
```
Comment (MikeMcQuaid) [inline Formula/pyenv.rb]: Inline this version.

File: Formula/pyenv.rb [hunk 2]
```diff
@@ -1,9 +1,30 @@
...
+        else
+          -1
+        end
+      elsif DATETIME_VERSIONS =~ other
+        1
+      else
+        super
+      end
+    end
+  end
```
Comment (MikeMcQuaid) [inline Formula/pyenv.rb]: We are unlikely to want to do this but will let @DomT4 chip in with thoughts.

Comment (yyuu) [inline Formula/pyenv.rb]: Thanks for reviewing!

**Comment by tdsmith** [comment]
An alternative: until we land proper version epoch support, we _could_ construct a synthetic version like `version "20160815.#{Version.parse url}"`

**Comment by MikeMcQuaid** [comment]
May be worth blocking this on https://github.com/Homebrew/homebrew-core/pull/3900.

**Comment by MikeMcQuaid** [comment]
May be worth blocking this on https://github.com/Homebrew/brew/pull/719.

**Comment by DomT4** [comment]
Agree with Mike's comment directly above.

**Comment by MikeMcQuaid** [comment]
@yyuu I'm hoping to get https://github.com/Homebrew/brew/pull/719 merged this week so this shouldn't be held up too much longer.

**Comment by yyuu** [comment]
Got it. I'll keep giving my eyes on the PR :grinning:

File: Formula/pyenv.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Formula/pyenv.rb]: You can remove all of this now.

File: Formula/pyenv.rb [hunk 3]
```diff
@@ -1,9 +1,29 @@
...
+
   desc "Python version management"
   homepage "https://github.com/yyuu/pyenv"
-  url "https://github.com/yyuu/pyenv/archive/v20160726.tar.gz"
-  sha256 "297040445f25912bea030be91b7d46c65eedbdf60655c2aae6226e88ed9b1346"
+  url "https://github.com/yyuu/pyenv/archive/v1.0.0.tar.gz"
+  sha256 "93216b9998cd1a3a09b8cc205a17de9a27b06788660749d36192c6a5ca346b4a"
   head "https://github.com/yyuu/pyenv.git"
+  version Pyenv::Version.new("1.0.0")
```
Comment (MikeMcQuaid) [inline Formula/pyenv.rb]: You can remove this line, add an empty line and `version_scheme 1` which will address this. The new docs in https://github.com/Homebrew/brew/pull/743 may help with understanding.

**Comment by MikeMcQuaid** [comment]
Actually, figured it'd be easier to create the PR myself in https://github.com/Homebrew/homebrew-core/pull/4006. Thanks for creating this one though, @yyuu ✨

**Pull request closed without merging**